test/etl_test.py

```diff
@@ -131,3 +130,93 @@ def test_ferc1_solo_etl(datastore_fixture,
+
+    def test_get_fields(self):
+        """Check that the get fields table works as expected."""
+        expect_path = pathlib.Path(__file__).parent / "data" / "ferc" \
+                      / "form1" / "f1_2018" / "get_fields.json"
+
+        with expect_path.open() as f:
+            expect = yaml.load(f.read(), yaml.Loader)
+
+        data = get_archive_file(2018, "F1_PUB.DBC", testing=True)
+        result = get_fields(data)
+        assert result == expect
+
+
+    def test_sample_get_dbc_map(self):
+        table = get_dbc_map(2018, testing=True)
+        assert table["f1_429_trans_aff"] == {
+            "ACCT_CORC": "acct_corc",
+            "ACCT_CORC_": "acct_corc_f",
+            "AMT_CORC": "amt_corc",
+            "AMT_CORC_F": "amt_corc_f",
+            "DESC_GOOD2": "desc_good_serv_f",
+            "DESC_GOOD_": "desc_good_serv",
+            "NAME_COMP": "name_comp",
+            "NAME_COMP_": "name_comp_f",
+            "REPORT_PRD": "report_prd",
+            "REPORT_YEA": "report_year",
+            "RESPONDENT": "respondent_id",
+            "ROW_NUMBER": "row_number",
+            "ROW_PRVLG": "row_prvlg",
+            "ROW_SEQ": "row_seq",
+            "SPPLMNT_NU": "spplmnt_num"
+        }
+
+class TestExcelExtractor:
+    """Verify that we can lead excel files as provided via the datastore."""
+
+    def test_excel_filename_eia860(self):
+        """Spot check eia860 extractor gets the correct excel sheet names."""
+        extractor = pudl.extract.eia860.Extractor()
+        assert extractor.excel_filename(
+            2011, "boiler_generator_assn") == "EnviroAssocY2011.xlsx"
+        assert extractor.excel_filename(
+            2016, "generator_retired") == "3_1_Generator_Y2016.xlsx"
+        assert extractor.excel_filename(
+            2018, "utility") == "1___Utility_Y2018.xlsx"
+
+    def test_excel_filename_eia923(self):
+        """Spot check eia923 extractor gets the correct excel sheet names."""
+        extractor = pudl.extract.eia923.Extractor()
+        assert extractor.excel_filename(2009, "plant_frame") == \
+            "EIA923 SCHEDULES 2_3_4_5 M Final 2009 REVISED 05252011.XLS"
+        assert extractor.excel_filename(2019, "energy_storage") == \
+            "EIA923_Schedules_2_3_4_5_M_11_2019_21JAN2020.xlsx"
+        assert extractor.excel_filename(2012, "puerto_rico") == \
+            "EIA923_Schedules_2_3_4_5_M_12_2012_Final_Revision.xlsx"
+
+    def test_extract_eia860(self):
+        """Spot check extraction of eia860 excel files."""
+        extractor = pudl.extract.eia860.Extractor()
+        assert "Utility" in extractor.load_excel_file(
+            2015, "utility", testing=True).sheet_names
+        assert "Retired and Canceled" in extractor.load_excel_file(
+            2012, "generator_proposed", testing=True).sheet_names
+        assert "OwnerY09" in extractor.load_excel_file(
+            2009, "ownership", testing=True).sheet_names
+
+    def test_extract_eia923(self):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/manual/zaneselvans/pudl/pullRequest?prid=5437498)